src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/UpdateQueueService.kt

```diff
@@ -19,42 +21,48 @@ class UpdateQueueService(
   private val prisonerSqsClient by lazy { prisonerQueue.sqsClient }
   private val prisonerQueueUrl by lazy { prisonerQueue.queueUrl }
 
-  fun sendMessage(context: VisitContext) {
+  fun sendMessage(context: Context) {
     val sqsMessage = PrisonerDomainEventsListener.SQSMessage(
       Type = "RETRY",
       Message = objectMapper.writeValueAsString(context),
-      MessageId = "retry-${context.vsipId}"
+      MessageId = "retry-${context.getId()}"
     )
     val result = prisonerSqsClient.sendMessage(
       SendMessageRequest(prisonerQueueUrl, objectMapper.writeValueAsString(sqsMessage))
     )
 
     telemetryClient.trackEvent(
-      "visit-booked-create-map-queue-retry",
-      mapOf("messageId" to result.messageId!!, "vsipId" to context.vsipId),
+      "create-map-queue-retry",
+      mapOf("messageId" to result.messageId!!, "id" to context.getId()),
     )
   }
+}
 
-  fun sendMessage(context: IncentiveContext) {
-    val sqsMessage = PrisonerDomainEventsListener.SQSMessage(
-      Type = "IncentiveRETRY",
-      Message = objectMapper.writeValueAsString(context),
-      MessageId = "retry-${context.incentiveId}"
-    )
-    val result = prisonerSqsClient.sendMessage(
-      SendMessageRequest(prisonerQueueUrl, objectMapper.writeValueAsString(sqsMessage))
-    )
+@JsonTypeInfo(
+  use = JsonTypeInfo.Id.NAME,
+  include = JsonTypeInfo.As.PROPERTY,
+  property = "type",
+  defaultImpl = VisitContext::class
+)
+@JsonSubTypes(
+  JsonSubTypes.Type(value = VisitContext::class, name = "VISIT"),
+  JsonSubTypes.Type(value = IncentiveContext::class, name = "INCENTIVE")
+)
+abstract class Context(val type: String) {
+  abstract fun getId(): String
+}
```

Comment:
This allows jackson to automatically deserialise the retry message to the right context subclass